apps/web/src/pages/ApplicationFormPage/ApplicationFormPage.tsx

```diff
@@ -0,0 +1,427 @@
+import { closestCenter, DndContext, DragEndEvent } from '@dnd-kit/core';
+import {
+  arrayMove,
+  SortableContext,
+  verticalListSortingStrategy,
+} from '@dnd-kit/sortable';
+import {
+  EducationOrEmploymentStatus,
+  Field,
+  FieldInCroatian,
+  FoundOutAboutInternshipBy,
+} from '@internship-app/types';
+import { Checkbox, FormControl, RadioGroup } from '@mui/joy';
+import Radio from '@mui/joy/Radio';
+import { Button, Input } from '@mui/material';
+import clsx from 'clsx';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { usePostIntern } from '../../api/usePostIntern';
+import { SortableField } from '../../components/SortableField';
+import classes from './index.module.css';
+
+export type FormValues = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  phoneNumber: number;
+  dateOfBirth: string;
+  fields: Field[];
+  educationOrEmploymentStatus: EducationOrEmploymentStatus;
+  highSchoolOrCollegeName: string;
+  foundOutAboutInternshipBy: FoundOutAboutInternshipBy;
+  reasonForApplying: string;
+};
+
+export const ApplicationFormPage = () => {
+  const [internFields, setInternFields] = useState<Field[]>([]);
+
+  const { mutate: createInternMutation } = usePostIntern();
+
+  const { register, handleSubmit, formState } = useForm<FormValues>({
+    defaultValues: {
+      firstName: '',
+      lastName: '',
+      email: '',
+      phoneNumber: 0,
+      dateOfBirth: '',
+      fields: [],
+      educationOrEmploymentStatus: EducationOrEmploymentStatus.Other,
+      highSchoolOrCollegeName: '',
+      foundOutAboutInternshipBy: FoundOutAboutInternshipBy.Friend,
+      reasonForApplying: '',
+    } as FormValues,
+  });
+
+  const { errors } = formState;
+
+  const onSubmit = (data: FormValues) => {
+    const internToAdd = {
+      firstName: data.firstName,
+      lastName: data.lastName,
+      email: data.email,
+      phoneNumber: data.phoneNumber,
+      dateOfBirth: data.dateOfBirth,
+      fields: internFields,
+      educationOrEmploymentStatus: data.educationOrEmploymentStatus,
+      highSchoolOrCollegeName: data.highSchoolOrCollegeName,
+      foundOutAboutInternshipBy: data.foundOutAboutInternshipBy,
+      reasonForApplying: data.reasonForApplying,
+    };
+
+    try {
+      createInternMutation(internToAdd);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Btw, MUI i RHF [ovo mogu sami](https://stackoverflow.com/questions/61273514/react-material-ui-how-to-use-select-with-multiple-values-inside-react-hook-f)

Reply:
✅